devel/sn-demangler/Portfile

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           java 1.0
+
+github.setup        indoorvivants sn-demangler 0.0.3 v
+revision            0
+
+checksums           rmd160  cfde95d48d2de051cbbbe44148163b769a92f9dc \
+                    sha256  73fdba83ee023a3dc2578b51f0be377c23b1948eb1ae8e43ca78fb702a1bbd4f \
+                    size    5209
+
+categories          devel
+platforms           darwin
+supported_archs     noarch
+license             Apache-2
+maintainers         {@catap korins.ky:kirill} openmaintainer
+
+description         Scala Native name demangler
+long_description    ${description}
+
+# Required java version
+java.version        1.8+
+# LTS JDK port to install if required java not found
+java.fallback       openjdk11
+
+use_configure       no
+
+compiler.whitelist  clang
```

Comment:
is this line needed - it doesn't work without doing anything like this explicitly? I suppose you want it to build with an Apple Clang compiler - isn't that the default anyway?

Reply:
okay, so the default is fine - it just wouldn't work if someone explicitly selected a `gcc` version to compile it. I think it's fine to use it like this (or you could blacklist gcc, not sure what's preferred). In any case, this should probably become then `*clang*` so that also `macports-clang` versions would be allowed (see [here](https://trac.macports.org/wiki/UsingTheRightCompiler))

Reply:
@reneeotten it is working on my machine and I haven't got installed any another clang ;)

Reply:
Yes, that's not all that surprising since an Apple clang is used by default... If the package doesn't want to compile with gcc, then I would suggest adding `compiler.blacklist  *gcc*` for completeness. In that case if someone would do `sudo port install sn-demangler configure.compiler=macports-gcc-11` it would error out early because the compiler is not supported.

Reply:
@reneeotten so, that should I do with this line?

Reply:
I gave my suggestion above, instead of whitelisting I'd recommend blacklisting gcc compilers

Reply:
After thinking about this for a while I preferer to move with `*clang*`. Thanks!

Reply:
@reneeotten I guess blacklist with `*gcc*` is fine. Because macports might have clang (a lot of them) and gcc. Let switch off only one which isn't supported.